Kernel/Arch/i386/CPU.h

```diff
@@ -421,7 +422,17 @@ struct [[gnu::aligned(16)]] FPUState
 
 inline constexpr u32 page_base_of(u32 address)
 {
-    return address & 0xfffff000;
+    return address & PAGE_MASK;
+}
+
+inline constexpr u32 offset_in_page(u32 address)
+{
+    return address & (~PAGE_MASK);
+}
+
+inline VirtualAddress create_vm_address(VirtualAddress v_base, PhysicalAddress p_offset)
+{
+    return VirtualAddress(page_base_of(v_base.get()) + offset_in_page(p_offset.get()));
 }
```

Comment:
This API mixes `VirtualAddress` and `PhysicalAddress`, when the purpose of those classes is to *not* mix virtual and physical address. :)

Use of this function can be replaced by existing `VirtualAddress` functionality:
```cpp
auto addr = v_base.page_base().offset(some_offset);
```

Reply:
Yeah, your argument of "offsetting an Address with another is not making sense" does make sense :)
Fixed.